src/Datadog.Trace.DuckTyping/DuckType.Properties.cs

```diff
@@ -287,6 +288,7 @@ private static MethodBuilder GetPropertySetMethod(TypeBuilder proxyTypeBuilder,
             }
 
             il.Emit(OpCodes.Ret);
+            _methodBuilderGetToken.Invoke(proxyMethod, null);
```

Comment:
Why are we eagerly trying to get the metadata token? Did this previously not create a metadata token yet?

Reply:
Some of the crashes we have seen contains messages like `System.MissingMethodException : Method not found: ' PropertyPublicObject.ShowEnum(TestEnum)'` when creating the final Type in the TypeBuilder. We are not really sure why this appears randomly. 

So trying to avoid this error we apply both: A global lock over the TypeBuilder when we are feeding it with all the MethodBuilders and Properties, and the second is to force the runtime to create the token (`CreateTokenNoLock`) for the method before trying to create the type.

I'm not 100% sure this actually fix the issue, but in case the runtime fails to create the token we crash earlier.